Utilities/Pipeline/pipeline.py

```diff
@@ -0,0 +1,79 @@
+class PipelineStrategy(object):
+    """
+    Blackbox for function to be executed, during execution it searches
+    for data point with given IDs, run the function and
+    returns own data point with given ID.
+    """
+
+    def __init__(self, function=None, inputs_IDs=[],
+                 output_ID=None):
+        if function:
+            self.function = function
+        self.inputs_IDs = inputs_IDs
+        self.output_ID = output_ID
+
+    def execute(self, input_list):
+        inputs = [input.data for input in
+                  find_inputs(input_list, self.inputs_IDs)]
+        data = self.function(inputs)
+        output = PipelineDataPoint(data, self.output_ID)
+        return output
+
+    def function(self):
+        print("Strategy execution inputs: {}, output: {}".format(
+            self.inputs_IDs, self.output_ID))
+        return PipelineDataPoint(self.inputs_IDs, self.output_ID)
+
+
+def find_inputs(list_of_inputs, inputs_IDs):
+    inputs = [input for input in list_of_inputs if input.id in inputs_IDs]
+    inputs = sorted(inputs, key=lambda input: input.id)
+    return inputs
+
+
+class PipelineDataPoint(object):
+    """
+    Basic data type in the pipeline, it is used as a container
+    to pass data between strategies.
+    """
+
+    def __init__(self, data=None, id=None):
+        if data is None and id is None:
```

Comment:
Will this trigger when I pass the id but don't pass the data?

Reply:
Fixed, both values should be passed.